core/components/organisms/verticalNav/Link.tsx

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+
+interface Props {
+  children: React.ReactNode;
+  componentType?: string;
+  className?: string;
+  onKeyDown?: (event: React.KeyboardEvent<HTMLLinkElement>) => void;
+}
+
+const GenericText = ({ children, componentType = 'a', className, ...props }: Props) => {
+  return React.createElement(componentType, { ...props, className }, children);
+};
```

Comment:
Instead of using `React.createElement` we can directly use the `<a>` anchor tag as keyDown event is only applicable for link element in this case